torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -0,0 +1,151 @@
+
+      auto rrefId = RRefId::fromIValue(src->retRRefId());
+      auto forkId = ForkId::fromIValue(src->retForkId());
+      TORCH_CHECK(rrefId != forkId, "Does not support remote call to self.");
+
+      auto& ctx = RRefContext::getInstance();
+      auto ownerRRef = ctx->getOrCreateOwnerRRef<IValue>(rrefId);
+
+      // TODO: make this asynchronous
+      // src is only alive within this block, use reference to avoid copy
+      auto& stack = src->stackRef();
+      src->op()->getOperation()(stack);
+      AT_ASSERT(
+          stack.size() == 1,
+          "Return value of a builtin operator or a "
+          "TorchScript function should be a single IValue, got a vector of "
+          "size ",
+          stack.size());
+
+      ownerRRef->setValue(std::move(stack.front()));
+      return nullptr;
+    }
+    case MessageType::RREF_FETCH_CALL: {
+      auto srf = static_cast<ScriptRRefFetchCall*>(rpc);
+      // TODO: make this asynchronous
+      std::shared_ptr<OwnerRRef<IValue>> rref =
+          RRefContext::getInstance()->getOrCreateOwnerRRef<IValue>(
+              RRefId::fromIValue(srf->value()));
+      return c10::guts::make_unique<ScriptRRefFetchRet>(rref->getValue());
+    }
+    case MessageType::RREF_USER_CREATE: {
+      auto sra = static_cast<ScriptRRefCreate*>(rpc);
+      RRefContext::getInstance()->addFork(sra->valueRef());
+      return nullptr;
+    }
+    case MessageType::RREF_USER_DELETE: {
+      auto srd = static_cast<ScriptRRefDelete*>(rpc);
+      RRefContext::getInstance()->delFork(srd->valueRef());
+      return nullptr;
+    }
+    case MessageType::MESSAGE_WITH_AUTOGRAD_REQ: {
+      auto rpcWithAutograd = static_cast<RpcWithAutograd*>(rpc);
+      const auto& autogradMetadata = rpcWithAutograd->autogradMetadata();
+
+      // Attach 'recv' autograd function.
+      DistAutogradContext* autogradContext = addRecvRpcBackward(
+          rpcWithAutograd->autogradMetadata(), rpcWithAutograd->tensors());
+
+      // Process the original RPC.
+      auto wrappedMessageType = rpcWithAutograd->wrappedMessageType();
+      auto wrappedRpcResponse = processRpc(
+          rpcWithAutograd->moveWrappedRpc().get(), wrappedMessageType);
+
+      // Wrap the response with autograd, need a new autograd message id for
+      // each send/recv pair.
+      auto& autogradContainer = DistAutogradContainer::getInstance();
+      AutogradMetadata responseAutogradMetadata(
+          autogradMetadata.autogradContextId,
+          autogradContainer.newAutogradMessageId());
+
+      auto response = c10::guts::make_unique<RpcWithAutograd>(
+          MessageType::MESSAGE_WITH_AUTOGRAD_RESP,
+          responseAutogradMetadata,
+          std::move(wrappedRpcResponse));
+
+      // Attach the 'send' autograd function if needed.
+      if (autogradContext != nullptr) {
+        addSendRpcBackward(
+            *autogradContext, responseAutogradMetadata, response->tensors());
+      }
+      return response;
```

Comment:
I'm getting a build error here when trying to build this diff locally (`cannot bind ‘std::unique_ptr<torch::distributed::rpc::RpcWithAutograd>’ lvalue to ‘std::unique_ptr<torch::distributed::rpc::RpcWithAutograd>&&’`). It seems that `return std::move(response)` fixes this.

Reply:
@rohan-varma do you still see the error with the latest version?

Reply:
@mrshenli I wasn't able to reproduce this error on a devserver (i was getting it on devgpu). I will check if I'm still getting this on latest